src/cmd/root.go

```diff
@@ -513,6 +529,71 @@ var rootCmd = &cobra.Command{
 				return newToolResult(result, nil)
 			})
 
+		// Register campaigns tool
+		s.AddTool(
+			mcp.NewTool(
+				"campaigns",
+				mcp.WithDescription("Get all the campaigns in the OpsLevel account. Campaigns are used to track and manage initiatives or projects within OpsLevel."),
+				mcp.WithString("status", mcp.Description("Filter campaigns by status"), mcp.Enum(opslevel.AllCampaignStatusEnum...)),
+				mcp.WithToolAnnotation(mcp.ToolAnnotation{
+					Title:           "Campaigns in OpsLevel",
+					ReadOnlyHint:    &trueValue,
+					DestructiveHint: &falseValue,
+					IdempotentHint:  &trueValue,
+					OpenWorldHint:   &trueValue,
+				}),
+			),
+			func(ctx context.Context, req mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+				args := &opslevel.ListCampaignsVariables{}
+				status := req.GetString("status", "")
+				if status != "" {
+					status_enum := opslevel.CampaignStatusEnum(status)
+					args.Status = &status_enum
+				}
+
+				resp, err := client.ListCampaigns(args)
+				if err != nil {
+					return mcp.NewToolResultErrorFromErr("failed to list campaigns", err), nil
+				}
+				var campaigns []serializedCampaign
+				for _, node := range resp.Nodes {
+					c := serializedCampaign{
+						Id:           string(node.Id),
+						Name:         node.Name,
+						ProjectBrief: node.ProjectBrief,
+						Status:       string(node.Status),
+						HtmlURL:      node.HtmlUrl,
+						CheckStats:   node.CheckStats,
+						ServiceStats: node.ServiceStats,
+					}
+					if node.Owner != (opslevel.TeamId{}) {
+						c.Owner = &node.Owner
+					}
+					if node.StartDate != (iso8601.Time{}) {
+						c.StartDate = &node.StartDate
+					}
+					if node.EndedDate != (iso8601.Time{}) {
+						c.EndedDate = &node.EndedDate
+					}
+					if node.TargetDate != (iso8601.Time{}) {
+						c.TargetDate = &node.TargetDate
+					}
+					if len(node.Reminder.Channels) > 0 {
+						c.Reminder = &node.Reminder
+					}
+					if node.Filter != (opslevel.FilterId{}) {
+						c.Filter = &node.Filter
+					}
+					if node.CheckStats != (opslevel.Stats{}) {
+						c.CheckStats = node.CheckStats
+					}
```

Comment:
I've never seen syntax like this before - TIL

Reply:
Maybe not very golike  but I don't know how else to see if the value is the same as the default constructor